src/admin_upload/views.py

```diff
@@ -0,0 +1,89 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, JsonResponse
+from django.contrib.auth.decorators import user_passes_test
+from geonode.layers.models import Dataset
+import json
+import psycopg2
+import sys
+from woolpert import settings
+
+def admin_required(function):
+    def wrap(request, *args, **kwargs):
+        user = request.user
+        if user.is_superuser:
+            return function(request, *args, *kwargs)
+        else:
+            return redirect("/")
+    wrap.__doc__ = function.__doc__
+    wrap.__name__ = function.__name__
+    return wrap
+
+# Create your views here.
+@admin_required
+def admin_form(request):
+    if(request.POST):
+        json_data = json.loads(request.POST.get('json_data'))
+        x_index = [i for i, x in enumerate(json_data["columns"]) if x == 'x_dd']
+        y_index = [i for i, x in enumerate(json_data["columns"]) if x == 'y_dd']
+
+        dbHost     = settings.DATABASE_HOST
+        dbName  = settings.GEONODE_GEODATABASE
+        dbPassword = settings.GEONODE_GEODATABASE_PASSWORD
+        dbUsername    = settings.GEONODE_GEODATABASE_USER
+
+        error_list = []
+
+        conn = psycopg2.connect(
+            database=dbName, user=dbUsername, password=dbPassword, host=dbHost, port="5432"
+        )
+        cursor = conn.cursor()
+        cursor.execute(f"SELECT max(fid) as max_id FROM {json_data['layer']}")
```

Comment:
We should not need this, when inserting records we should always skip the fid or primary key. Ping me if you want to know more

Reply:
@NyakudyaA I had an issue where the primary key in my database was not auto incremented

Reply:
You can simply do the following if your PK is not auto incrementing
```
create sequence measures_measure_id_seq
   owned by measures.measure_id;

alter table measures
   alter column measure_id set default nextval('measures_measure_id_seq');
```